File: bronze/2231/python/main.py

if (__name__ == "__main__"):
    n = int(input())
    

    for i in range(int(n/2), n):
        sum = i
        temp = i
        while (temp):
            sum += temp % 10
            temp = int(temp / 10)
        if (sum == n):
            print(i)
            exit(0)
        i = temp
    print(0)

File: bronze/2231/c++/main.cpp
#include <iostream>

int main()
{
    
    int n;
    std::cin >> n;
    int sum, temp;
    for (int i = n/2; i < n; i++) {
        temp = i;
        sum = i;
        while (temp) {
            sum += temp % 10;
            temp /= 10;
        }

        if (n == sum) {
            std::cout << i;
            return 0;
        }
    }
    std::cout << 0;

    return 0;
}